hpke.go

```diff
@@ -184,21 +184,38 @@ func keySchedule(suite CipherSuite, mode Mode, sharedSecret, info, psk, pskID []
 	return params, nil
 }
 
+// Context represents an HPKE context encoded on the wire.
+type Context struct {
+	Role                 Role
+	KemId, KdfId, AeadId uint16
+	ExporterSecret       []byte `tls:"head=1"`
+	Key                  []byte `tls:"head=1"`
+	Nonce                []byte `tls:"head=1"`
+	Seq                  uint64
+}
+
+// Role specifies the role of a party in possession of a Context: if equal to
+// `0`, then the party is the sender; if equal to `0`, then the party is the
+// receiver.
+type Role uint8
```

Comment:
For readability, I would move this above `Context`

Reply:
This doesn't need to be a public type, does it?  That is, it could be `role`.

Reply:
Well, since `Role` is an exported field of `Context`, I think it also makes sense to export its type. I'm happy to go with your suggestion, however.

Reply:
I don't see why Context needs to be exported, since you can't get to it anyway.

Reply:
Hmmm... I suppose the variants should be exported as well? I.e., "roleSender" -> "RoleSender".

Reply:
If we're not exporting `Context`, then I see no reason to export `Role`. I'll make this change.